src/internal/utils/screen.ts

```diff
@@ -1,18 +1,78 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-export function getNormalizedElementRect(element: HTMLElement): DOMRect {
-  const { x, y, left, right, top, bottom, width, height } = element.getBoundingClientRect();
+import { PointerEvent as ReactPointerEvent } from "react";
+
+export function getNormalizedElementRect(element: HTMLElement): {
+  left: number;
+  right: number;
+  top: number;
+  bottom: number;
+  width: number;
+  height: number;
+} {
+  const { left, right, top, bottom, width, height } = getLogicalBoundingClientRect(element);
   const xOffset = element.ownerDocument.defaultView!.pageXOffset - window.scrollX;
   const yOffset = element.ownerDocument.defaultView!.pageYOffset - window.scrollY;
   return {
-    x: x + xOffset,
     left: left + xOffset,
     right: right + xOffset,
-    y: y + yOffset,
     top: top + yOffset,
     bottom: bottom + yOffset,
     width: width,
     height: height,
-  } as DOMRect;
```

Comment:
The DOMRect type is used in a bunch of places, it would be nice to replace it with some custom Rect type featuring rtl-friendly property names, maybe as a separate PR.